lightning/src/offers/invoice_request.rs

```diff
@@ -335,52 +338,70 @@ impl<'a, 'b, P: PayerIdStrategy, T: secp256k1::Signing> InvoiceRequestBuilder<'a
 		self
 	}
 
-	pub(super) fn build_unchecked(self) -> UnsignedInvoiceRequest<'a> {
+	pub(super) fn build_unchecked(self) -> UnsignedInvoiceRequest {
 		self.build_without_checks().0
 	}
 }
 
 /// A semantically valid [`InvoiceRequest`] that hasn't been signed.
-pub struct UnsignedInvoiceRequest<'a> {
-	offer: &'a Offer,
-	invoice_request: InvoiceRequestContents,
+pub struct UnsignedInvoiceRequest {
+	bytes: Vec<u8>,
```

Comment:
just to make sure that I understand, there's no way to canonically go from `contents` to the `bytes` (to skip lugging them around)?